.github/workflows/rules.yml

```diff
@@ -5,13 +5,17 @@ env:
   CUDA_PATH:
 
 on:
+  workflow_dispatch:
   push:
   pull_request:
     types: [labeled]
 
 jobs:
   build:
-    if: ${{ contains(github.event.pull_request.labels.*.name, 'run-workflow') || github.event_name == 'push' }} && env.CUDA_PATH != ''
+    env:
+      var_a: ${{ env.CUDA_PATH != ''}}
```

Comment:
You should not set in the `env` section of this job, but actually use another job and set inside the steps.

Otherwise, you are still probing the same as before.
It is definitely more readable, but the effect has not changed.

Reply:
It is not strictly needed, because we could avoid running tests with GPU requirements altogether.

In any case, if we'll ever have a self-hosted runner, it will receive a separate workflow. So, I believe you could throw the `CUDA_PATH` variable in the workflow altogether, and just apply the Pytest filtering.

Do you agree @liweintu?

Reply:
Modified env section and updated the file.